liblouis/compileTranslationTable.c

```diff
@@ -2707,12 +2889,101 @@ compileRule(FileInfo *file, TranslationTableHeader **table,
 		}
 		return putCharDotsMapping(
 				file, ruleChars.chars[0], ruleDots.chars[0], displayTable);
+	case CTO_None: {
+		// check if token is a macro name
+		if (inScopeMacros) {
+			const MacroList *macros = *inScopeMacros;
+			while (macros) {
+				const Macro *m = macros->head;
+				if (token.length == strlen(m->name) &&
+						eqasc2uni((unsigned char *)m->name, token.chars, token.length)) {
+					if (!inScopeMacros) {
+						compileError(file, "Calling macros only allowed in table files.");
+						return 0;
+					}
```

Comment:
Is this code even reachable? Above you say `if (inScopeMacros) ` why do you need to test for `if (!inScopeMacros)` here?

Reply:
This is where we compile a macro, whereas the other code is where we compile a rule that uses the macro. Just different things.